pkg/cli/mirror/copy_test.go

```diff
@@ -0,0 +1,372 @@
+package mirror
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/openshift/library-go/pkg/image/reference"
+	"github.com/openshift/oc-mirror/pkg/image"
+	"github.com/operator-framework/operator-registry/alpha/model"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	testdata        = "testdata/artifacts/rhop-ctlg-oci"
+	rotten_manifest = "testdata/artifacts/rhop-rotten-manifest"
+	rotten_layer    = "testdata/artifacts/rhop-rotten-layer"
+	rotten_config   = "testdata/artifacts/rhop-rotten-cfg"
+	other_layer     = "testdata/artifacts/rhop-not-catalog"
+)
+
+func TestParse(t *testing.T) {
+	toTest := "quay.io/skhoury/ocmir/albo/aws-load-balancer-controller-rhel8@sha256:d7bc364512178c36671d8a4b5a76cf7cb10f8e56997106187b0fe1f032670ece"
+	s, err := reference.Parse(toTest)
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
+	rf, err := image.ParseReference(toTest)
+
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
+	fmt.Printf("%s - %s\n", s, rf)
+}
+
+// TODO: add preparation step that saves a catalog locally before testing
+// see maybe contents of pkg/image/testdata
+func TestGetOCIImgSrcFromPath(t *testing.T) {
+	type spec struct {
+		desc  string
+		inRef string
+		err   string
+	}
+	wdir, err := os.Getwd()
+	if err != nil {
+		t.Fatal("unable to get working dir")
+	}
+	cases := []spec{
+		{
+			desc:  "full path passes",
+			inRef: filepath.Join(wdir, testdata),
+			err:   "",
+		},
+		{
+			desc:  "relative path passes",
+			inRef: testdata,
+			err:   "",
+		},
+		{
+			desc:  "inexisting path should fail",
+			inRef: "/inexisting",
+			err:   "unable to get OCI Image from /inexisting: open /inexisting/index.json: no such file or directory",
```

Comment:
This causes the unit test to fail, since the error string is actually `unable to get OCI Image from oci:/inexisting: open /inexisting/index.json: no such file or directory`.

Reply:
Fixed
Thanks!